src/cider/nrepl/middleware/debug.clj

```diff
@@ -0,0 +1,67 @@
+(ns cider.nrepl.middleware.debug
+  (:require [clojure.tools.nrepl.transport :as transport]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.misc :refer [response-for]]
+            [debugger.core :refer [break]]))
+
+;;;; ## Interaction with the client
+;;; `wrap-debug` receives an initial message from the client, stores
+;;; it in `debugger-message`, and `breakpoint` answers it.
+(def debugger-message nil)
+
+(defn wrap-debug [h]
+  (fn [{:keys [op force] :as msg}]
+    (if (= "init-debugger" (:op msg))
+      (if (and debugger-message (not force))
+        (transport/send (:transport msg)
+                        (response-for msg :status :done))
+        (do (when debugger-message
```

Comment:
Ok. I think I understand now. You are using `debugger-message` simply as a storage for the most recent `debug-init` request (given that  `cider-debug-defun-at-point` is calling `cider--debug-init-connection`).